### PR TITLE
Improved RoleClass System

### PR DESCRIPTION
**Pull request by Ultradragon005**
Thx EHR for the basecode.

This is done to make it less convoluted and faster.

### CODE REVIEW

File: Roles/Core/CustomRoleManager.cs [hunk 1]
```diff
@@ -13,57 +13,15 @@ namespace TOHE.Roles.Core;
 public static class CustomRoleManager
 {
     public static readonly Dictionary<CustomRoles, RoleBase> RoleClass = [];
+    public static RoleBase GetRoleClass(this CustomRoles role) => RoleClass.TryGetValue(role, out var roleClass) & roleClass != null ? roleClass : new VanillaRole(); // Main alr assigns it new vanilla if null, but incase it is addon somehow O-o
```
Comment (Tommy-XL) [inline Roles/Core/CustomRoleManager.cs:16]: Better name it like `GetRoleClassByRole()`

**Comment by Gurge44** [comment]
couldn't even spell EHR correctly 😕

**Comment by Ultradragon005** [comment]
Tomato, Tomato, it was a honest mistake <:hehe:1114307136026050650>

**Comment by Tommy-XL** [comment]
Fix conflicts pls

**Comment by Ultradragon005** [comment]
fixed

**Pull request merged**